packages/core/src/browser/tree/tree.ts

```diff
@@ -67,8 +67,9 @@ export interface ITreeNode {
      */
     readonly description?: string;
     /**
-     * Test whether this node is visible.
-     * If undefined then visible.
+     * Test whether this node should be rendered.
+     * If undefined then node will be rendered.
+     * Note, this flag doesn't reflect actual visibility on UI.
```

Comment:
The last sentence should be removed: UI should reflect the tree state, not another way around. The tree model does not care about UI.